src/Aspire.Cli/Commands/InitCommand.cs

```diff
@@ -0,0 +1,229 @@
+        IPackagingService packagingService,
+        AspireCliTelemetry telemetry,
+        IDotNetSdkInstaller sdkInstaller,
+        IFeatures features,
+        ICliUpdateNotifier updateNotifier,
+        CliExecutionContext executionContext,
+        IInteractionService interactionService)
+        : base("init", InitCommandStrings.Description, features, updateNotifier, executionContext, interactionService)
+    {
+        ArgumentNullException.ThrowIfNull(runner);
+        ArgumentNullException.ThrowIfNull(certificateService);
+        ArgumentNullException.ThrowIfNull(prompter);
+        ArgumentNullException.ThrowIfNull(templateFactory);
+        ArgumentNullException.ThrowIfNull(packagingService);
+        ArgumentNullException.ThrowIfNull(telemetry);
+        ArgumentNullException.ThrowIfNull(sdkInstaller);
+
+        _runner = runner;
+        _certificateService = certificateService;
+        _prompter = prompter;
+        _templateFactory = templateFactory;
+        _packagingService = packagingService;
+        _telemetry = telemetry;
+        _sdkInstaller = sdkInstaller;
+        _features = features;
+        _updateNotifier = updateNotifier;
+        _executionContext = executionContext;
+
+        var sourceOption = new Option<string?>("--source", "-s");
+        sourceOption.Description = NewCommandStrings.SourceArgumentDescription;
+        sourceOption.Recursive = true;
+        Options.Add(sourceOption);
+
+        var templateVersionOption = new Option<string?>("--version", "-v");
+        templateVersionOption.Description = NewCommandStrings.VersionArgumentDescription;
+        templateVersionOption.Recursive = true;
+        Options.Add(templateVersionOption);
+    }
+
+    protected override async Task<int> ExecuteAsync(ParseResult parseResult, CancellationToken cancellationToken)
+    {
+        // Check if the .NET SDK is available
+        if (!await SdkInstallHelper.EnsureSdkInstalledAsync(_sdkInstaller, InteractionService, cancellationToken))
+        {
+            return ExitCodeConstants.SdkNotInstalled;
+        }
+
+        using var activity = _telemetry.ActivitySource.StartActivity(this.Name);
+
+        // Check for solution files (.sln or .slnx) in the current directory
+        var solutionFile = FindSolutionFile(_executionContext.WorkingDirectory);
+
+        if (solutionFile is not null)
+        {
+            InteractionService.DisplayMessage("information", string.Format(CultureInfo.CurrentCulture, InitCommandStrings.SolutionDetected, solutionFile.Name));
+            return await InitializeExistingSolutionAsync(solutionFile, parseResult, cancellationToken);
+        }
+        else
+        {
+            InteractionService.DisplayMessage("warning", InitCommandStrings.NoSolutionFoundCreatingSingleFileAppHost);
+            return await CreateSingleFileAppHostAsync(parseResult, cancellationToken);
+        }
+    }
+
+    private static FileInfo? FindSolutionFile(DirectoryInfo directory)
```

Comment:
Walk up directores looking for a solution file (closest is the most relevant). If you find more than one, prompt for ambiguity and give the users a choice to pick which one they want to add the projects to.

Look at ProjectLocator for inspiration on how to implement this search. We may want to introduce `SolutionLocator` with simialr semantics.